src/pages/index.js

```diff
@@ -56,31 +78,85 @@ function handleImageClick(data) {
 }
 
 function handleEditProfileFormSubmit(inputValues) {
-  userInfo.setUserInfo({
-    name: inputValues.title,
-    job: inputValues.description,
-  });
-  editProfileModal.close();
+  const { title, description } = inputValues;
+  editProfileModal.setSaving(true);
+  api
+    .updateProfile(title, description)
+    .then((data) => {
+      userInfo.setUserInfo({
+        name: data.name,
+        job: data.about,
+      });
+      editProfileModal.close();
+    })
+    .catch(handleError)
+    .finally(() => editProfileModal.setSaving(false));
+}
+
+function handleDeleteCard(card) {
+  {
+    confirmPopup.setSubmitFunction(() => {
+      api
+        .deleteCard(card._id)
+        .then(() => {
+          card.remove();
+          confirmPopup.close();
+        })
+        .catch(handleError);
+    });
+
+    confirmPopup.open();
+  }
 }
 
 function handleAddCardFormSubmit(inputValues) {
+  addCardModal.setSaving(true);
   const cardData = { name: inputValues.title, link: inputValues.url };
-  const newCard = createCard(cardData);
-  cardSection.addItem(newCard);
-  addCardModal.close();
-  modalAddForm.reset(); // clear the form
-  addCardFormValidator.disableButton(); // disable the button
+  api
+    .addCard(cardData)
+    .then((newCard) => {
+      const cardElement = createCard(newCard);
+      cardSection.addItem(cardElement);
+      addCardModal.close();
+      modalAddForm.reset();
+      addCardFormValidator.disableButton();
+      addCardModal.setSaving(true);

```

Comment:
`addCardModal.setSaving(true);` is already in line 113 when you **start** loading. It's not needed in `.then`

Reply:
* My remark hasn't been corrected https://github.com/Jonathan-Brandt/se_project_aroundtheus/pull/1#discussion_r1955527258  


Please, delete `addCardModal.setSaving(true);` from line 115 because it's not needed in `.then`. You have it in line 105 already